sdk/search/azure-search/pom.xml

```diff
@@ -37,6 +37,16 @@
       <version>1.4.0-beta.2</version> <!-- {x-version-update;unreleased_com.azure:azure-core-http-netty;dependency} -->
     </dependency>
 
+    <!-- Added this dependency to include necessary annotations used by reactor core.
+    Without this dependency, javadoc throws a warning as it cannot find enum When.MAYBE
+    which is used in @Nullable annotation in reactor core classes -->
+    <dependency>
+      <groupId>com.google.code.findbugs</groupId>
+      <artifactId>jsr305</artifactId>
+      <version>3.0.2</version> <!-- {x-version-update;com.google.code.findbugs:jsr305;external_dependency} -->
+      <scope>provided</scope>
+    </dependency>
```

Comment:
This is already added in azure-core. Do we need to add this here as well?

Reply:
`provided` dependencies don't appear to flow transitively to downstream modules, this is needed for `azure-search`.